modules/apps/27-interchain-accounts/controller/ibc_middleware.go

```diff
@@ -176,6 +176,61 @@ func (im IBCMiddleware) OnTimeoutPacket(
 	return im.app.OnTimeoutPacket(ctx, packet, relayer)
 }
 
+// OnChanUpgradeInit implements the IBCMiddleware interface
+func (im IBCMiddleware) OnChanUpgradeInit(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID, channelID string,
+	sequence uint64,
+	counterparty channeltypes.Counterparty,
```

Comment:
@AdityaSripal referring to our conversation yesterday, should `counterparty`  be one of the mutable values?

Reply:
What do you mean mutable? This cannot and will not change across an upgrade.

But I'm not sure if it would ever be useful for an application to have during an upgrade callback.

cc: @colin-axner @damiannolan

Reply:
i meant that the other values here might change across an upgrade, but this should not